# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/vm/JSObject-inl.h
# Commit: 4531d5c25694
# Full Hash: 4531d5c256947ff782377ba0143e3f37bd7207c2
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2021-03-06 09:33:16
# Regressor Bug: 1689413
# File Overlap Count: 1
# Description:
#   Bug 1689413 part 5 - Remove JSClass* from ObjectGroup. r=jonco
#   
#   BaseShape stores the JSClass* too so this patch changes the code to get the JSClass*
#   from there instead of from the ObjectGroup.
#   
# ==============================================================================

diff -r f1bb16079c0e -r 4531d5c25694 js/src/vm/JSObject-inl.h
--- a/js/src/vm/JSObject-inl.h	Fri Mar 05 19:10:06 2021 +0000
+++ b/js/src/vm/JSObject-inl.h	Fri Mar 05 19:10:06 2021 +0000
@@ -362,11 +362,6 @@
   return gc::DefaultHeap;
 }
 
-inline gc::InitialHeap GetInitialHeap(NewObjectKind newKind,
-                                      ObjectGroup* group) {
-  return GetInitialHeap(newKind, group->clasp());
-}
-
 /*
  * Make an object with the specified prototype. If parent is null, it will
  * default to the prototype's global if the prototype is non-null.